### PR TITLE
Enhanced documentation for Array#repeated_permutation

### PR DESCRIPTION
**Pull request by BurdetteLamar**

### CODE REVIEW

File: array.c [hunk 1]
```diff
@@ -7810,25 +7810,77 @@ rb_ary_repeated_permutation_size(VALUE ary, VALUE args, VALUE eobj)
...
- *     a.repeated_permutation(0).to_a  #=> [[]] # one permutation of length 0
+ *  Returns a new \Enumerator if no block given:
+ *    a = [0, 1, 2]
+ *    a.repeated_permutations(2) # => #<Enumerator: [0, 1, 2]:permutation(2)>
+ *
+ *  Using Enumerators, it's convenient to show the permutations and counts
+ *  for some values of +n+:
+ *    e = a.repeated_permutation(0)
+ *    e.size # => 1
+ *    e.to_a # => 27
```
Comment (jeremyevans) [inline array.c:7868]: This should be `# => [[]]`

File: array.c [hunk 2]
```diff
@@ -7810,25 +7810,77 @@ rb_ary_repeated_permutation_size(VALUE ary, VALUE args, VALUE eobj)
...
+ *    e = a.repeated_permutation(1)
+ *    e.size # => 1
```
Comment (jeremyevans) [inline array.c:7870]: This should be `# => 3`

File: array.c [hunk 3]
```diff
@@ -7810,25 +7810,77 @@ rb_ary_repeated_permutation_size(VALUE ary, VALUE args, VALUE eobj)
...
+ *    e.to_a # => 27
```
Comment (jeremyevans) [inline array.c:7871]: This should be `# => [[0], [1], [2]]`

File: array.c [hunk 4]
```diff
@@ -7810,25 +7810,77 @@ rb_ary_repeated_permutation_size(VALUE ary, VALUE args, VALUE eobj)
...
+ *    e = a.repeated_permutation(2)
+ *    e.size # => 9
+ *    e.to_a # => [[0, 0], [0, 1], [0, 2], [1, 0], [1, 1], [1, 2], [2, 0], [2, 1], [2, 2]]
+ *    e = a.repeated_permutation(3)
```
Comment (jeremyevans) [inline array.c:7875]: I don't think we need a fourth example here.

File: array.c [hunk 5]
```diff
@@ -7810,25 +7810,77 @@ rb_ary_repeated_permutation_size(VALUE ary, VALUE args, VALUE eobj)
...
+ *    e.size # => 27
+ *    e.to_a # => [[0, 0, 0], [0, 0, 1], [0, 0, 2], [0, 1, 0], [0, 1, 1], [0, 1, 2], [0, 2, 0], [0, 2, 1], [0, 2, 2], [1, 0, 0], [1, 0, 1], [1, 0, 2], [1, 1, 0], [1, 1, 1], [1, 1, 2], [1, 2, 0], [1, 2, 1], [1, 2, 2], [2, 0, 0], [2, 0, 1], [2, 0, 2], [2, 1, 0], [2, 1, 1], [2, 1, 2], [2, 2, 0], [2, 2, 1], [2, 2, 2]]
+ *
+ *  ---
+ *
+ *  Raises an exception if +n+ is not an Integer-convertible object:
```
Comment (jeremyevans) [inline array.c:7881]: I realize similar things were documented for other methods, but this is implied by the statement above that `n` must be Integer-convertible.  However, if you want to change everything later after the method documentation guide is approved, you can keep it for now.

File: array.c [hunk 6]
```diff
@@ -7810,25 +7810,77 @@ rb_ary_repeated_permutation_size(VALUE ary, VALUE args, VALUE eobj)
...
+ *    [0, 2]
+ *    [1, 0]
+ *    [1, 1]
+ *    [1, 2]
+ *    [2, 0]
+ *    [2, 1]
+ *    [2, 2]
+ *
+ *  If +n+ is zero, calls the block once with an empty \Array:
+ *    a.repeated_permutation(0) {|permutation| p permutation }
```
Comment (jeremyevans) [inline array.c:7852]: Not sure we need an example here, the statement is self explanatory.

File: array.c (hunk 4 above)
Comment (sergioro9) [reply array.c:7875]: ## this iis a text

`and another text`

just testing github markdown

File: array.c (hunk 1 above)
Comment (BurdetteLamar) [reply array.c]: Fixed.

File: array.c (hunk 2 above)
Comment (BurdetteLamar) [reply array.c]: Fixed.

File: array.c (hunk 3 above)
Comment (BurdetteLamar) [reply array.c]: Fixed.

File: array.c (hunk 4 above)
Comment (BurdetteLamar) [reply array.c]: Removed.

File: array.c (hunk 5 above)
Comment (BurdetteLamar) [reply array.c:7878]: Stet.

**Comment by BurdetteLamar** [comment]
Thanks, @jeremyevans, for being careful.  (I was not.)

**Pull request merged**